modules/core/src/main/java/org/apache/ignite/internal/lang/IgniteExceptionMapperUtil.java

```diff
@@ -115,7 +136,7 @@ public static Throwable mapToPublicException(Throwable origin) {
         }
 
         // There are no exception mappings for the given exception. This case should be considered as internal error.
-        return new IgniteException(INTERNAL_ERR, origin);
+        return unknownProblemMapper.apply(origin);
```

Comment:
Should we check that `unknownProblemMapper` returns an instance of the `public` exception?

Reply:
Added an assertion